### DISCUSSION TITLE
Active desktop switches to Desktop 1 after every screenshot on MacOS

### DISCUSSION THREAD

**Discussion by mikiTesf** [flameshot-org/flameshot#2874] [Q&A]
I have been using flameshot on Ubuntu for the past 2 years but recently had to switch to MacOS. I remember that on Linux, after taking a screenshot all the flameshot UI disappears and I get to keep a nice screenshot in the clipboard and continue on the same window/desktop I was on. After the switch to MacOS (Monterey 12.5) immediately after copying the screenshot flameshot switches the active desktop to `Desktop 1`. I'm running this 👉 `v12.1.0 (96c2c82e)` version of flameshot. Is this how it used to be or a bug introduced in a recent release?

**Comment by bunk3m** [comment]
I just upgraded to Monterey over the weekend.  Prior to the upgrade, I ran Flameshot via Macports and it worked great.  With the upgrade, I started using the app downloaded here.  Unfortunately with the app version I am getting exactly what @mikiTesf  is seeing.  When I use the quick command shift+command+x to capture a screenshot, Flameshot ends up switching to another desktop which has only the background.  I'm not sure if there are any specific settings I need to change as everything is default.  Attached screenshot shows what I get now.

Any suggestion for whichever settings I need to change are appreciated.  I suspect this is a setting issue but as @mikiTesf said, might be a bug due to Monterey.
<img width="1440" alt="Screen Shot 2022-08-24 at 08 56 52" src="https://user-images.githubusercontent.com/3494567/186425293-e34c8abe-23fb-418b-8284-eba41b1be901.png">

**Comment by bunk3m** [comment]
This additional screenshot might help describe what is going on. See the strange "flameshot" desktop/environment at the far right?
<img width="933" alt="Screen Shot 2022-08-24 at 22 52 05" src="https://user-images.githubusercontent.com/3494567/186563576-decf681d-557a-4b92-88fc-e3d19683916e.png">

**Comment by bunk3m** [comment]
BTW, I installed from sources via Macports to test another installed version of the software.  The same result.  I now think it's a bug.

**Comment by bunk3m** [comment]
Created a bug.

**Reply by mikiTesf** [reply]
Please follow the format [here](https://github.com/flameshot-org/flameshot/issues/new?assignees=&labels=Unconfirmed+Bug&template=bug-report.yaml) when creating bug cases.

**Reply by mmahmoudian** [reply]
@mikiTesf Thank you for helping us and other users. It is really inconvenient at best, and painful at worst when we get bug reports that lacks the information we need.

@bunk3m few points:
1. as far as I can understand, your issue, and the one @mikiTesf explained are different.
2. in macOS, when you make any software full screen (Chrome, Finder, Terminal, ...) they create their own desktop (try it and you'll see). This is how macOS works. Now, the reason you don't see the things you want in the screenshot is because your permissions are not set. delete the existing permissions of Flameshot and add them again. This is a very annoying bug that macOS has. Follow insteructions here: https://flameshot.org/docs/guide/troubleshooting/#macos
3. you are not alone, but everyone have solved this the way I explained: https://github.com/flameshot-org/flameshot/issues?q=is%3Aissue+is%3Aclosed+label%3AMacOS+label%3ADuplicate

**Comment by mmahmoudian** [comment]
@mikiTesf 
>  immediately after copying the screenshot flameshot switches the active desktop to Desktop 1

Do you mean that you can take a good screenshot, but you were sent to Desktop1 instead of where you were before (e.g Desktop2)?

**Reply by mikiTesf** [reply]
Exactly.

**Reply by mmahmoudian** [reply]
So all the stuff you want to take screenshot from are visible in the final screenshot, it is just you are sent to a wrong desktop after that?

**Reply by mikiTesf** [reply]
Yes. The screenshot function is fine and dandy. All the modifications (edits) I make to the screenshot are all there but immediately when I save/copy it I find myself on `Desktop1` instead of the fullscreen window I took the screenshot from. I have to swipe my way back there.

**Reply by mmahmoudian** [reply]
Okay, thanks, now I understand the issue better. This is a discussion board, so other devs might miss the discussions here. I suggest opening a bug report with this beautiful and clear description you just gave. Then it gets the visibility and can be viewed and discussed by all the devs and potential contributors.

**Reply by borgmanJeremy** [reply]
Thanks for the chat here, agree on opening a bug report

**Reply by bunk3m** [reply]
@mikiTesf you are lucky. I get to the blank desktop but can't take a screenshot because what I wanted to screenshot is on another desktop.

**Reply by mikiTesf** [reply]
> Okay, thanks, now I understand the issue better. This is a discussion board, so other devs might miss the discussions here. I suggest opening a bug report with this beautiful and clear description you just gave. Then it gets the visibility and can be viewed and discussed by all the devs and potential contributors.

https://github.com/flameshot-org/flameshot/issues/2886

**Reply by mikiTesf** [reply]
> @mikiTesf you are lucky. I get to the blank desktop but can't take a screenshot because what I wanted to screenshot is on another desktop.

Did you try resetting the permission and rebooting?

**Reply by bunk3m** [reply]
Yes multiple times.  No luck.
